internal/consensus/state.go

```diff
@@ -2264,54 +2311,59 @@ func (cs *State) addVote(vote *types.Vote, peerID p2p.ID) (added bool, err error
 		"extLen", len(vote.Extension),
 		"extSigLen", len(vote.ExtensionSignature),
 	)
+}
 
-	if vote.Height < cs.Height || (vote.Height == cs.Height && vote.Round < cs.Round) {
-		cs.metrics.MarkLateVote(vote.Type)
-	}
+// isLateVote checks if a vote is a late vote.
+func (cs *State) isLateVote(vote *types.Vote) bool {
```

Comment:
This is harder to read than the original version.

The same for the similar helpers. I think that the content of the ifs should be maintained for better readability. A comment can do the work of explaining what is being evaluated.